rio_cogeo/cogeo.py

```diff
@@ -93,6 +104,41 @@ def cog_translate(
             if alpha:
                 vrt_params.update(dict(add_alpha=False))
 
+            if web_optimized:
+                bounds = list(
+                    transform_bounds(
+                        *[src_dst.crs, "epsg:4326"] + list(src_dst.bounds),
+                        densify_pts=21
+                    )
+                )
+                center = [(bounds[0] + bounds[2]) / 2, (bounds[1] + bounds[3]) / 2]
+
+                tilesize = min(
+                    int(dst_kwargs["blockxsize"]), int(dst_kwargs["blockysize"])
+                )
+                lat = 0 if latitude_adjustment else center[1]
+                max_zoom = get_max_zoom(src_dst, lat=lat, tilesize=tilesize)
+
+                extrema = tile_extrema(bounds, max_zoom)
+                w, n = mercantile.xy(
+                    *mercantile.ul(extrema["x"]["min"], extrema["y"]["min"], max_zoom)
+                )
+                vrt_res = _meters_per_pixel(max_zoom, 0, tilesize=tilesize)
+                vrt_transform = Affine(vrt_res, 0, w, 0, -vrt_res, n)
+
+                vrt_width = (extrema["x"]["max"] - extrema["x"]["min"]) * tilesize
+                vrt_height = (extrema["y"]["max"] - extrema["y"]["min"]) * tilesize
+
+                vrt_params.update(
+                    dict(
+                        crs="epsg:3857",
+                        transform=vrt_transform,
+                        width=vrt_width,
+                        height=vrt_height,
+                        resampling=Resampling[overview_resampling],
```

Comment:
Well in fact this is not ready! 
I added a little shortcut for the resampling method but this should be a proper options! (🎉 one more options)